File: src/main/java/com/chattriggers/ctjs/minecraft/wrappers/objects/PotionEffect.java
//package com.chattriggers.ctjs.minecraft.wrappers.objects;
//
////#if MC>10809
////$$ import net.minecraft.potion.Potion;
////#endif
//
//public class PotionEffect {
//    private net.minecraft.potion.PotionEffect effect;
//
//    public PotionEffect(net.minecraft.potion.PotionEffect effect) {
//        this.effect = effect;
//    }
//
//    /**
//     * @return the name of the potion effect
//     */
//    public String getName() {
//        return this.effect.getEffectName();
//    }
//
//    /**
//     * @return the amplifier of the potion effect
//     */
//    public int getAmplifier() {
//        return this.effect.getAmplifier();
//    }
//
//    /**
//     * @return the duration of the potion effect
//     */
//    public int getDuration() {
//        return this.effect.getDuration();
//    }
//
//    /**
//     * @return the ID of the potion effect
//     */
//    public int getID() {
//        //#if MC<=10809
//        return this.effect.getPotionID();
//        //#else
//        //$$ return Potion.getIdFromPotion(this.effect.getPotion());
//        //#endif
//    }
//
//    /**
//     * @return True if the potion effect is ambient
//     */
//    public Boolean isAmbient() {
//        return this.effect.getIsAmbient();
//    }
//
//    /**
//     * @return True if the potion effect is the max duration
//     */
//    public Boolean isDurationMax() {
//        return this.effect.getIsPotionDurationMax();
//    }
//
//    /**
//     * @return True if the potion effect is showing particles
//     */
//    public Boolean showsParticles() {
//        //#if MC<=10809
//        return this.effect.getIsShowParticles();
//        //#else
//        //$$ return this.effect.doesShowParticles();
//        //#endif
//    }
//
//    public String toString() {
//        return this.effect.toString();
//    }
//}

File: src/main/kotlin/com/chattriggers/ctjs/minecraft/wrappers/objects/PotionEffect.kt
package com.chattriggers.ctjs.minecraft.wrappers.objects

import com.chattriggers.ctjs.minecraft.objects.message.TextComponent
import com.chattriggers.ctjs.utils.kotlin.External
import net.minecraft.entity.effect.StatusEffectInstance
import net.minecraft.text.TranslatableText
import net.minecraft.util.registry.Registry

@External
class PotionEffect(val effect: StatusEffectInstance) {
    /**
     * Returns the translation key of the potion.
     * Ex: "potion.poison"
     */
    fun getName(): String = effect.translationKey

    /**
     * Returns the localized name of the potion that
     * is displayed in the player's inventory.
     * Ex: "Poison"
     */
    fun getLocalizedName(): String = TextComponent(TranslatableText(getName())).getFormattedText()

    fun getAmplifier(): Int = effect.amplifier

    fun getDuration(): Int = effect.duration

    fun getID() = Registry.STATUS_EFFECT.getRawId(effect.effectType)

    fun isAmbient(): Boolean = effect.isAmbient

    // TODO("fabric")
    // fun isDurationMax(): Boolean = effect.isPotionDurationMax

    fun showsParticles() = effect.shouldShowParticles()

    override fun toString(): String = effect.toString()
}
